lib/src/ui/puzzle/puzzle_history_screen.dart

```diff
@@ -0,0 +1,295 @@
+            MediaQuery.of(context).size.width > kTabletThreshold ? 4 : 2;
+        final boardWidth = (defaultTargetPlatform == TargetPlatform.iOS)
+            ? (MediaQuery.of(context).size.width) * 0.91 / crossAxisCount
+            : MediaQuery.of(context).size.width / crossAxisCount;
+        return ListSection(
+          header: Text(context.l10n.puzzleHistory),
+          headerTrailing: (data.historyList.length == 10)
+              ? NoPaddingTextButton(
+                  onPressed: () => pushPlatformRoute(
+                    context,
+                    builder: (context) => PuzzleHistoryScreen(data.historyList),
+                  ),
+                  child: Text(
+                    context.l10n.more,
+                  ),
+                )
+              : null,
+          children: [
+            for (var i = 0; i < data.historyList.length; i += crossAxisCount)
+              Row(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: data.historyList
+                    .getRange(
+                      i,
+                      i + crossAxisCount <= data.historyList.length
+                          ? i + crossAxisCount
+                          : data.historyList.length,
+                    )
+                    .map(
+                      (e) => _HistoryBoard(e, boardWidth),
+                    )
+                    .toList(),
+              ),
+          ],
+        );
+      },
+      error: (e, s) {
+        debugPrint(
+          'SEVERE: [PuzzleHistoryWidget] could not load dashboard',
+        );
+        return const Center(child: Text('Could not load Puzzle History'));
+      },
+      loading: () => Shimmer(
+        child: ShimmerLoading(
+          isLoading: true,
+          child: ListSection.loading(
+            itemsNumber: 5,
+            header: true,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class PuzzleHistoryScreen extends StatelessWidget {
+  const PuzzleHistoryScreen(this.historyList);
+
+  final IList<HistoryPuzzle> historyList;
+
+  @override
+  Widget build(BuildContext context) {
+    return PlatformWidget(androidBuilder: _buildAndroid, iosBuilder: _buildIos);
+  }
+
+  Widget _buildIos(BuildContext context) {
+    return CupertinoPageScaffold(
+      navigationBar: CupertinoNavigationBar(
+        middle: Text(context.l10n.puzzleHistory),
+      ),
+      child: _Body(historyList),
+    );
+  }
+
+  Widget _buildAndroid(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text(context.l10n.puzzleHistory)),
+      body: _Body(historyList),
+    );
+  }
+}
+
+class _Body extends ConsumerStatefulWidget {
+  const _Body(this.historyList);
+  final IList<HistoryPuzzle> historyList;
+  @override
+  ConsumerState<_Body> createState() => _BodyState();
+}
+
+class _BodyState extends ConsumerState<_Body> {
+  final ScrollController _scrollController = ScrollController();
+  List<HistoryPuzzle> _historyList = [];
```

Comment:
So yes it's weird here to have both a `provider` and `state`. The provider isn't actually used as a provider but rather as a repository. And the state is in the stateful widget. It should be one or another but not both.